src/Board/Board.php

```diff
@@ -18,20 +21,32 @@ public function __construct()
         $rows    = static::getRows();
         $white   = true;    // Is the tile white? We'll flip this to false to mean black.
         
-        foreach ($columns as $column) {
-            foreach ($rows as $row) {
+        foreach ($rows as $row) {
+            foreach ($columns as $column) {
                 $colour    = $white ? 'white' : 'black';
-                $reference = $column . $row;
+                $reference = $row . $column;
                 $tile      = new Tile($colour, $reference);
                 
-                if (2 === $row) {
+                if (2 === $column) {
                     $tile->setPiece(new Pawn('white'));
-                } elseif (7 === $row) {
+                } elseif (7 === $column) {
                     $tile->setPiece(new Pawn('black'));
                 } elseif (('C1' === $reference) || ('F1' === $reference)) {
                     $tile->setPiece(new Bishop('white'));
                 } elseif (('C8' === $reference) || ('F8' === $reference)) {
                     $tile->setPiece(new Bishop('black'));
+                } elseif (('A1' === $reference) || ('H1' === $reference)){
+                    $tile->setPiece(new Rook('white'));
+                } elseif (('A8' === $reference) || ('H8' === $reference)){
+                    $tile->setPiece(new Rook('black'));
+                } elseif ('E1' === $reference) {
+                    $tile->setPiece(new King('white'));
+                } elseif ('E8' === $reference) {
+                    $tile->setPiece(new King('black'));
+                } elseif ('D1' === $reference){
```

Comment:
Could combine these conditions, then use ```$colour``` in the constructor of the piece, e.g.

```
elseif ('D1' === $reference || 'D8' === $reference) {
    $tile->setPiece(new Queen($colour));
}
```

This also goes for the existing code for Pawn and Bishop pieces.

Though, these multiple conditions could become quite long, so could use ```in_array()``` instead of combining ors:
```
} elseif (in_array($reference, ['C1', 'F1', 'C8', 'F8'], true)) {
     $tile->setPiece(new Bishop($colour));
}
```